# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1233258758 18000
#      Thu Jan 29 14:52:38 2009 -0500
# Node ID cbd6545b0d85b21b1b9b1914ce34ba1034092c12
# Parent  0c5541c39a421f823ad604266c2d21f853988d19
fileparts.m: match all possible file separators

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-01-29  John W. Eaton  <jwe@octave.org>
+
+	* miscellaneous/fileparts.m: Match all possible file separators.
+
 2009-01-29  Kai Habel <kai.habel@gmx.de>
 
         * plot/__go_draw_axes__.m: Add support for transparent patches.
 
 2009-01-29  Søren Hauberg  <hauberg@gmail.com>
 
 	* help/help.m, help/print_usage.m, help/get_first_help_sentence.m:
 	print sensible error message when function is found but not documented.
diff --git a/scripts/miscellaneous/fileparts.m b/scripts/miscellaneous/fileparts.m
--- a/scripts/miscellaneous/fileparts.m
+++ b/scripts/miscellaneous/fileparts.m
@@ -22,17 +22,20 @@
 ## @var{filename}.
 ## @seealso{fullfile}
 ## @end deftypefn
 
 function [directory, name, extension, version] = fileparts (filename)
 
   if (nargin == 1)
     if (ischar (filename))
-      ds = rindex (filename, filesep);
+      ds = strchr (filename, filesep ("all"), 1, "last");
+      if (isempty (ds))
+	ds = 0;
+      endif
       es = rindex (filename, ".");
       ## These can be the same if they are both 0 (no dir or ext).
       if (es <= ds)
 	es = length(filename)+1;
       endif
       if (ds == 0)
 	directory = "";
       elseif (ds == 1)
